### ISSUE TITLE
Playback fails to start on DLNA Player provider with JRiver Media Center renderer

### ISSUE THREAD

**Issue by madbrain76** [music-assistant/support#2493]
### What version of Music Assistant has the issue?

2.0.7

### What version of the Home Assistant Integration have you got installed?

2024.6.2

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Unable to play content from music providers onto JRiver media center using DLNA

### How to reproduce

1. install JRiver Media center on a host of your choice . I use version 31 on Windows
2. In MC, go to Tools / Options / Media Network and turn on "Enable Media Network" (first checkbox)
3. In MA, install the DLNA/UPNP player provider
4. select the player corresponding to the JRiver instance. In my case, it was "HIGGS"
5. select a track or album and hit play
6. Note that playback does not start in MC
7. Go back to JRiver MC
8. Click on the main zone (HIGGS in my case)
9. Notice that the track selected in step 5 shows up in the bottom pane, with all metadata including artwork
10. Press play in JRiver
11. The track starts playing in MC
12. Stop playback from the MC side

### Music Providers

Plex

However, I tried other music providers before and the behavior was the same.

### Player Providers

DLNA/UPNP

### Full log output

This first log is includes steps 1-6 .

[log.txt](https://github.com/user-attachments/files/15860207/log.txt)

The second log goes all the way to the end, ie. step 12 .
[log2.txt](https://github.com/user-attachments/files/15860213/log2.txt)

I'm also including logs from JRiver MC in case they can help diagnose/fix the issue. First log is for steps 1-6.
[JRiver Log 2024-06-16 16-44-16.zip](https://github.com/user-attachments/files/15860215/JRiver.Log.2024-06-16.16-44-16.zip)

Second log is for steps 7-12 .
[JRiver Log 2024-06-16 16-45-26.zip](https://github.com/user-attachments/files/15860216/JRiver.Log.2024-06-16.16-45-26.zip)

### Additional information

Getting this to work would be quite useful. JRiver is able to expose nearly any Windows audio device(s) through DLNA, and thus would make it available to MA. JRiver runs on Linux and Mac too, but I haven't experimented with them (have Linux, no Mac).
If this works, it can also alleviate the need for a browser-based player provider when one is using a Windows/Mac/Linux system.

### What version of Home Assistant Core are your running

2024.6.2

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Windows

**Comment by OzGav** [comment]
keeping open for now

**Comment by marcelveldt** [comment]
bot shut up

**Comment by madbrain76** [comment]
FYI, I just checked this with MA 2.2.2, and the behavior is unchanged.

**Comment by OzGav** [comment]
So just confirming that this does work it is just that the initial play command fails and you have to press play in JRiver Media Center?

**Comment by madbrain76** [comment]
Yes, it's the startup that fails.  However, if you pause using MA, you also can't resume playback from MA. You must go back to the JRiver GUI to resume. I even tried to using MO4Media, a separate DLNA remote, and it wouldn't allow resumption in this case, either.

Interestingly, however, once I start playback with the JRiver, UI, I can use MO4media to pause/resume successfully, as long as I don't touch any transport controls in MA.

**Comment by madbrain76** [comment]
There are probably other DLNA bugs here. I start a playlist. At the end of the first track, it did not switch to the next track. Nothing I tried succeeded in making it play the second track. If I start the playlist in MA, hit play in JRiver, I'm able to use the MA prev/next transport controls to navigate between tracks, though.

**Comment by sjhilton1978** [comment]
Hi - I'm also trying to use JRiver MC via DLNA (I use MA to access Tidal and then serve that to an MC client via DLNA). I am having slightly different issues. I've been doing some digging on the JRiver forums and it appears that JRiver MC does not like 'chunking' when streaming FLAC by DLNA and needs a file with a defined length. I tried some of the different profiles in MA and had a bit of success getting it going using 'Profile 3 - forced content length'. The only issue is that MA seems to consistently switch to the next track before completion. I hope that assists.

**Comment by OzGav** [comment]
Is this still a problem with 2.4.4?

**Comment by sjhilton1978** [comment]
Yes it is from my testing

**Comment by sjhilton1978** [comment]
I think I've worked out what the issue is. It appears to be the way jriver handles flac via dlna. Switching music assistant from FLAC output to wav seems to work for me. I'd be interested to hear if that works for others.

**Comment by OzGav** [comment]
If this is a confirmed workaround we can add it to the docs

**Comment by OzGav** [comment]
This is pretty stale. I will add the workaround to the docs. This can be reopened in the future if there is anything more to add

**Comment by madbrain76** [comment]
> Yes it

Unfortunately, I cannot confirm the workaround. I tried setting both WAV uncompressed in the DLNA player, as well as Profile 3 - forced content length. I am not seeing any change in behavior. The JRiver side still updates the URL for the content from MA, but always fails to play, when initiated through MA.

There are a lot of variables involved, though, such as which version of MC you succeeded with, which music provider you are using in MA, what type of file is being played, and how your DLNA renderer is configured in JRiver MC.

For me, the answers are that I'm using the Qobuz provider. I believe 100% of the files stored on it are FLAC. My DLNA renderer is using the MC default for "Audiophile 24-bit DAC", which should force PCM 24-bit to be sent, without any format conversion on the MC side. I also tried "Generic DLNA" which should be more compatible, and allow for conversion. Neither worked.

Could you please detail the circumstances under which you got it to work successfully ?

**Comment by sjhilton1978** [comment]
Hi Julien - I can set it out in more detail. Before I do that I recommend
checking if you have precision zone sync turned on in jriver (under
settings, general, features). That was causing issues for me this week and
blocking music assistant playback altogether. I understand that uses non
standard dlna fields. If that doesn't work happy to provide more details on
my setup.

Stuart Hilton
***@***.***

**Comment by madbrain76** [comment]
> Hi Julien - I can set it out in more detail. Before I do that I recommend checking if you have precision zone sync turned  on in jriver (under settings, general, features). That was causing issues for me this week and blocking music assistant playback altogether. I understand that uses non standard dlna fields. If that doesn't work happy to provide more details on my setup.

Thanks. I just disabled it, but it made no difference. Please provide your setup details.

**Comment by sjhilton1978** [comment]
You do have to restart jriver to fully disable it. It didn't prompt me to
restart when I disabled it.

Stuart Hilton
***@***.***

**Comment by madbrain76** [comment]
> You do have to restart jriver to fully disable it. It didn't prompt me to
> restart when I disabled it.
> 
> Stuart Hilton
> ***@***.***
> […](#)

It did warn me to restart, and I did so, but again, this setting made no difference. Please provide your details.

**Comment by sjhilton1978** [comment]
I'm using Tidal as the provider. I think they're FLAC as well. I have Music Assistant set to WAV and Profile 2 No Content Length. I also have all of the sample rates/bitdepths selected although that shouldn't matter. On the Jriver side if I play something to my specific music assistant zone then music plays immediately, tracks progress as normal, but Jriver shows the feed as being 'Live' (might have something to do with the content length setting). For the DLNA settings it's Original Format with no modifications as far as I can tell.

**Comment by madbrain76** [comment]
> I'm using Tidal as the provider. I think they're FLAC as well. I have Music Assistant set to WAV and Profile 2 No Content Length. I also have all of the sample rates/bitdepths selected although that shouldn't matter. On the Jriver side if I play something to my specific music assistant zone then music plays immediately, tracks progress as normal, but Jriver shows the feed as being 'Live' (might have something to do with the content length setting). For the DLNA settings it's Original Format with no modifications as far as I can tell.

Thank you so much. I tried to use the same settings, but with Qobuz, but unfortunately, I'm still not able to make it start playing without interacting with the JRiver GUI.

As far as the file format, MA will show you the format if you click on a track and look at "Provider details" below the artwork. JRiver MC will also show the file format. Both sides are showing FLAC for my Qobuz tracks. Can you please double check that your tracks are FLAC also ? I believe they should be all FLAC on Tidal indeed, as you say.

I could perhaps do another free trial of Tidal to try to duplicate your environment, but before I do that, it would be to know.
a. the version of JRiver MC you are using. I'm on MC 31.0.87 for x64 Windows.
b. the version of Music Assistant you are using. I'm using on 2.6.0 beta 6 .

**Comment by sjhilton1978** [comment]
No worries - my MA is showing FLAC input and WAV output via DLNA. I'm running version 2.5.2 (stable). I'm running a linux Jriver v33.0.72 (latest/last version) (haven't moved to v34 yet). It's showing WAV playback. I wonder if there could be an issue with Qobuz? It seems odd that DLNA is not being presented with WAV if that's what you've selected?

**Comment by madbrain76** [comment]
> No worries - my MA is showing FLAC input and WAV output via DLNA. I'm running version 2.5.2 (stable). I'm running a linux Jriver v33.0.72 (latest/last version) (haven't moved to v34 yet). It's showing WAV playback. I wonder if there could be an issue with Qobuz? It seems odd that DLNA is not being presented with WAV if that's what you've selected?

Thank you very much for those details !

Yes, it's weird that the file format was being shown as flac still. I think there might be a delay for new DLNA player settings in MA to take effect, or perhaps it's a bug in the beta. Not quite sure.

I spent a lot of time testing various things - including playing a WAV file directly using the File system provider, trying MA 2.5.2 GA, various combinations of file format, content length, etc. Absolutely no combination I tried could force playback to start from the MA side onto my JRiver MC31 for Windows. I also have an MC31 for Linux, and the result was the same.

I then installed a trial of MC34 on Windows. I had to skip MC33, since I already did a trial, but the 30 days expired without me upgrading. At that point, I was able to get playback to start from MA. With this version of MC, the DLNA player settings in MA can be left as default - ie. the format can be left as FLAC. However, playback still doesn't work out of the box, until turning off Precision zone sync in MC. With MC34, that appears to be the only thing that needs to be documented. It looks like JRiver made some fixes to DLNA MC after MC31 that help with the compatibility with MA. I don't know if they were made in MC32 or MC34. And there could have been further changes in MC34 as well. Can you please try again with the DLNA output left as default (FLAC) and see if you are able to play onto your MC33, with Precision time sync off ? If so, that would be identical behavior to MC34.
I'm not sure what else JRiver can still fix in order to make things work seamlessly out of the box. I suppose changing Precision time sync to be off by default in MC would do it, but they might not want to do that.

**Comment by madbrain76** [comment]
Things were playing fine for a long time, maybe an hour, but at some point, the sound turned to that of a jackhammer in a loop. Pause / resume still produced the same sound. After pausing and waiting a while (several minutes) then resuming, the audio came back to normal. Not sure if that's a bug in the beta. Have you heard this before ?

[music-assistant.log](https://github.com/user-attachments/files/20270573/music-assistant.log)

**Comment by OzGav** [comment]
No that has not been reported. That would be a separate issue report if it can be reproduced. I am hoping we can add a note in the docs about need MC33 or higher and that setting off to at least put this one to bed.

**Comment by OzGav** [comment]
Is it confirmed that FLAC will playback on version 33 or later?

**Comment by madbrain76** [comment]
> No that has not been reported. That would be a separate issue report if it can be reproduced. I am hoping we can add a note in the docs about need MC33 or higher and that setting off to at least put this one to bed.

Yes, I know. I just want to see if anyone following this bug ever ran into it. There is nothing usable in the log attached. I ran into the jackhammer noise with JRiver MC one time, but when MC was the player, onto a WiiM device over DLNA, while transcoding DSD to 192 kHz PCM realtime. Reproducible after a few hours of playback. This sounds similar, except this is the reverse case, with MA being the player and JRiver MC the renderer. Unfortunately, the JRiver and WiiM are closed source DLNA unlike MA, and I didn't get anywhere reporting that bug to them last time. I'll head to the JRiver forums again when I get some time. And maybe WiiM forums too.
